controllers/dataplane/openstackdataplanedeployment_controller.go

```diff
@@ -443,5 +444,6 @@ func (r *OpenStackDataPlaneDeploymentReconciler) SetupWithManager(mgr ctrl.Manag
 				predicate.AnnotationChangedPredicate{},
 				predicate.LabelChangedPredicate{}))).
 		Owns(&ansibleeev1.OpenStackAnsibleEE{}).
+		Owns(&certmgrv1.Certificate{}).
```

Comment:
I think the ownership should be on the nodeset (but I'm not sure how the deployment would behave)

Reply:
but its the deployment which creates the the certs, so it owns it https://github.com/openstack-k8s-operators/openstack-operator/blob/main/controllers/dataplane/openstackdataplanedeployment_controller.go#L217

Reply:
or the nodeset needs a watcher based on label

Reply:
ah wait you are right. it passes the nodeset and its the owner. I'll update

Reply:
I'm thinking about the owner deletion, when the deployment is deleted, we don't want the certs deleted, but when the nodeset gets deleted, it makes sense to delete the certs.